src/Build/BuildCheck/OM/EvaluatedPropertiesAnalysisData.cs

```diff
@@ -12,9 +12,16 @@ namespace Microsoft.Build.Experimental.BuildCheck;
 public class EvaluatedPropertiesAnalysisData : AnalysisData
 {
     internal EvaluatedPropertiesAnalysisData(
-        string projectFilePath,
-        IReadOnlyDictionary<string, string> evaluatedProperties) :
-        base(projectFilePath) => EvaluatedProperties = evaluatedProperties;
+            string projectFilePath,
+            IReadOnlyDictionary<string, string> evaluatedProperties,
+            IReadOnlyDictionary<string, (string EnvVarValue, string File, int Line, int Column)> evaluatedEnvironmentVariables)
+            : base(projectFilePath)
+    {
+        EvaluatedProperties = evaluatedProperties;
+        EvaluatedEnvironmentVariables = evaluatedEnvironmentVariables;
+    }
 
     public IReadOnlyDictionary<string, string> EvaluatedProperties { get; }
+
+    public IReadOnlyDictionary<string, (string EnvVarValue, string File, int Line, int Column)> EvaluatedEnvironmentVariables { get; }
```

Comment:
I know the rest of file doesn't have comments. But this is going to be a public API - let's xml comment